src/components/MarkdownProvider/components/Typography.tsx

```diff
@@ -12,14 +12,17 @@ import { XXL, XL, LG } from '@zendeskgarden/react-typography';
 
 const headerSpacing = (p: ThemeProps<DefaultTheme>) => {
   return css`
-    margin-top: ${p.theme.space.md};
+    margin-top: ${p.theme.space.lg};
     margin-bottom: ${p.theme.space.sm};
-    padding-bottom: ${p.theme.space.xs};
+    font-weight: ${p.theme.fontWeights.semibold};
   `;
 };
 
 export const StyledH2 = styled(XXL).attrs({ tag: 'h2' })`
   ${headerSpacing}
+  margin-top: ${p => p.theme.space.xxl};
+  border-bottom: solid 1px ${p => getColor('grey', 300, p.theme)};
```

Comment:
Still looking for `p.theme.borders.sm` usage here instead of `solid 1px`.

Reply:
@rossmoody how did we decide collectively to add border to the main body's typography? Underlined / bordered section headers are rarely used in branded areas and do not appear in product. I understand borders are commonly represented in markdown headers (e.g. github style) and that you've advocated for this in the past. However, this is a major departure for Zendesk, and we should at least have a conversation about why you think this is necessary, and potentially explore other alternatives that are more on-brand (spacing, sizing, weight). Overall, I do not think this change is ready for production, as cases where lots of underlines will appear in close proximity have clearly not been considered: 

![image](https://user-images.githubusercontent.com/1230973/81869407-dac81e80-9528-11ea-99f5-6f03546a2a3a.png)